app/services/UserService.php

```diff
@@ -1,13 +1,16 @@
 <?php
-require_once __DIR__ . '/../repositories/UserRepository.php';
 
 class UserService
 {
     private $userRepository;
+    private $badgeRepository;
```

Comment:
Why do we need badgeRepository both here and in badgeservice ?

Reply:
because we moved some functions from badgeService to userService, since we don't want to have badgService instantiated in CocktailController

the functions moved to userService related to badges, require access to badgeRepository.

Reply:
If we have private badgeService instead of badgeRepository, we create more cohesive userService that combines the user data and their badges.

Reply:
in that case a function like badge notification will have the following flow
badgeRepository->badgeService->userService->basecontroller->cocktailController

Reply:
What do you mean?  BadgeRepository is a data access layer that intereacts with the database, but badge service is a business logic layer that uses badgeRepository already, I think that userService should not directly interact with the badgeRepository because that is just mixing responsabilities